### PR TITLE
Added function to return pretty names on "system" module "process summary" metricset

### PR DESCRIPTION
**Pull request by sayden**
Following this discuss https://discuss.elastic.co/t/errors-in-metric-beat-log/147918/3 It's true that I found that the error message was a bit cryptic.

According to this http://thelinuxstuff.blogspot.com/2012/08/process-state-codes-in-ps-output.html the state 88, which corresponds to `X` state in a linux process corresponds to `dead` state.

I thought it could be nice to have a bit more descriptive error message in such case to help user debugging his own problems:

So, in the case of the discuss issue where the message had this format:
```
Unknown state <88> for process with pid 30794
```
It will become this:

```
Unknown or unexpected state <X> (dead) for process with pid 30794
```

BTW, I couldn't manage to compile any test, system was complaining about some redefined variables:

```
GOROOT=/usr/lib/go #gosetup
GOPATH=/home/caster/go #gosetup
/usr/lib/go/bin/go test -c -i -tags "darwin freebsd linux windows" -o /tmp/___TestFetch_in_github_com_elastic_beats_metricbeat_module_system_process_summary github.com/elastic/beats/metricbeat/module/system/process_summary #gosetup
# runtime
/usr/lib/go/src/runtime/lock_sema.go:30:20: active_spin redeclared in this block
	previous declaration at /usr/lib/go/src/runtime/lock_futex.go:30:20
/usr/lib/go/src/runtime/lock_sema.go:31:20: active_spin_cnt redeclared in this block
	previous declaration at /usr/lib/go/src/runtime/lock_futex.go:31:20
/usr/lib/go/src/runtime/lock_sema.go:32:20: passive_spin redeclared in this block
	previous declaration at /usr/lib/go/src/runtime/lock_futex.go:32:20
/usr/lib/go/src/runtime/lock_sema.go:35:14: lock redeclared in this block
	previous declaration at /usr/lib/go/src/runtime/lock_futex.go:46:14
/usr/lib/go/src/runtime/lock_sema.go:94:16: unlock redeclared in this block
	previous declaration at /usr/lib/go/src/runtime/lock_futex.go:106:16
/usr/lib/go/src/runtime/lock_sema.go:124:19: noteclear redeclared in this block
	previous declaration at /usr/lib/go/src/runtime/lock_futex.go:126:19
/usr/lib/go/src/runtime/lock_sema.go:128:20: notewakeup redeclared in this block
	previous declaration at /usr/lib/go/src/runtime/lock_futex.go:130:20
/usr/lib/go/src/runtime/lock_sema.go:151:19: notesleep redeclared in this block
	previous declaration at /usr/lib/go/src/runtime/lock_futex.go:139:19
/usr/lib/go/src/runtime/lock_sema.go:180:68: notetsleep_internal redeclared in this block
	previous declaration at /usr/lib/go/src/runtime/lock_futex.go:164:45
/usr/lib/go/src/runtime/lock_sema.go:263:36: notetsleep redeclared in this block
	previous declaration at /usr/lib/go/src/runtime/lock_futex.go:210:36
/usr/lib/go/src/runtime/lock_sema.go:263:36: too many errors
```

### CODE REVIEW

**Comment by jsoriano** [comment]
If it is possible to have "dead" processes, as it seems by the discuss thread and [this stackexchange comment](https://unix.stackexchange.com/questions/126101/possible-to-see-dead-proccesses#comment202211_126207), I'd propose to count or ignore them and in any case don't report them as unexpected state to don't flood logs with something that is not really an error.

File: metricbeat/module/system/process_summary/process_summary.go [hunk 1]
```diff
@@ -96,7 +96,7 @@ func (m *MetricSet) Fetch() (common.MapStr, error) {
 		case 'Z':
 			summary.zombie++
 		default:
-			logp.Err("Unknown state <%v> for process with pid %d", state.State, pid)
+			logp.Err("Unknown or unexpected state <%c> (%s) for process with pid %d", state.State, prettyNameOf(state.State), pid)
```
Comment (jsoriano) [inline metricbeat/module/system/process_summary/process_summary.go]: I agree with the change to print the character value, but I'd keep this error log line only for unknown values, and not for not-so-expected states as could be the dead one.

**Comment by jsoriano** [comment]
Regarding the error on tests, how are you running them? this seems to be adding all platforms as tags, what will include the implementations for all platforms, this is why you see redefined declarations. You should be able to run the tests for this metricset with
```
go test ./metricbeat/module/system/process_summary/...
```

**Comment by sayden** [comment]
Found the solution to the test error (in the provided log you can see that it's trying to compile it to all platforms, I just needed to only set the platform `linux`). It was Goland that sets a lot of flags when running tests through the UI. Running them *the easy way* through the command line works fine.

Ok, so:

* We stay with the character instead of the ASCII number.
* I wouldn't hide the log. I understand that it is a hassle to flood the logs with this messages but it also helps users with debugging.
* `dead` is, currently, an unexpected process. I wouldn't like to hide it (means, to say that it's *unknown*). Instead I'd prefer to count it too and report it but this causes a change in the `common.MapStr` schema that I don't know if it can affect somewhere (I guess it won't but I prefer to be cautious)

**Comment by jsoriano** [comment]
> I wouldn't hide the log. I understand that it is a hassle to flood the logs with this messages but it also helps users with debugging.

I meant to hide the log for the case of `dead`, that is a possible known case, for unknown cases we should continue logging them. My concern is that we are logging as error something that is not an error.

> dead is, currently, an unexpected process. I wouldn't like to hide it (means, to say that it's unknown). Instead I'd prefer to count it too and report it but this causes a change in the common.MapStr schema that I don't know if it can affect somewhere (I guess it won't but I prefer to be cautious)

Yes, this was one of my proposals, counting them as any other state, but in any case a process is this state is going to disappear very soon, so I find it good too to just ignore them (that is not the same as unknown, because I agree that unknown states should be logged).

Regarding changes in the fields, adding them don't break compatibility.

File: metricbeat/module/system/process_summary/process_summary.go [hunk 2]
```diff
@@ -109,6 +112,7 @@ func (m *MetricSet) Fetch() (common.MapStr, error) {
 		"stopped":  summary.stopped,
 		"zombie":   summary.zombie,
 		"unknown":  summary.unknown,
+		"dead":     summary.dead,
```
Comment (jsoriano) [inline metricbeat/module/system/process_summary/process_summary.go]: You will need to add this new field to `metricbeat/module/system/process_summary/_meta/fields.yml`, after adding it you will also need to run `make update`.

Comment (jsoriano) [inline metricbeat/module/system/process_summary/process_summary.go]: Please, add also a changelog entry regarding this new field.

Comment (sayden) [reply metricbeat/module/system/process_summary/process_summary.go]: Done in e4c9bab8e90c4151813763c544210eb2a08f491e

Comment (sayden) [reply metricbeat/module/system/process_summary/process_summary.go]: Done in e4c9bab8e90c4151813763c544210eb2a08f491e

File: CHANGELOG.asciidoc [hunk 3]
```diff
@@ -119,6 +119,7 @@ https://github.com/elastic/beats/compare/v6.4.0...master[Check the HEAD diff]
 - Add experimental socket summary metricset to system module {pull}6782[6782]
 - Increase ignore_above for system.process.cmdline to 2048. {pull}8101[8100]
 - Add support to renamed fields planned for redis 5.0. {pull}8167[8167]
+- Added 'died' PID state to process_system metricset on system module
```
Comment (jsoriano) [inline CHANGELOG.asciidoc]: Add also a reference to the pull request as in other entries.

Comment (jsoriano) [reply CHANGELOG.asciidoc]: And you will also need to update the branch with master, there is a conflict in the changelog.

Comment (sayden) [reply CHANGELOG.asciidoc]: Fixed both in 503c282cb245dd8cdc9a10b5aca862bc852c6abc

**Pull request merged**